snippets/ResultValue.hpp

```diff
@@ -122,7 +122,11 @@
   struct ResultValueType
   {
 #ifdef VULKAN_HPP_NO_EXCEPTIONS
+#ifdef VULKAN_HPP_EXPECTED
+  using type = VULKAN_HPP_EXPECTED<T, Result>;
+#else
 	using type = ResultValue<T>;
+#endif
```

Comment:
That is, the type is either `VULKAN_HPP_EXPECTED<T, Result>` or `ResultValue<T>`, mostly depending on the C++ version?
Do those two types look similar enough, that your compiler won't realize a difference where that type is actually used? If not, your code base would fail when switching to C++23.
That is, maybe you need some additional `VULKAN_HPP_USE_STD_EXPECTED`, or so, as an additional guard in `snippets/MacrosHppTemplate.hpp`?